helix-core/src/main/java/org/apache/helix/model/IdealState.java

```diff
@@ -55,7 +55,9 @@
     MAX_PARTITIONS_PER_INSTANCE,
     INSTANCE_GROUP_TAG,
     REBALANCER_CLASS_NAME,
-    HELIX_ENABLED
+    HELIX_ENABLED,
+    RESOURCE_GROUP_NAME,
+    RESOURCE_GROUP_ENABLED
```

Comment:
Why do we need ResourceGroupEnabled flag? Will things work as expected if there is a resourcegroupName and by default we can set the resourceGroupName to resourceName ?